pysat/_instrument.py

```diff
@@ -1065,18 +1069,41 @@ def __setitem__(self, key, new_data):
                         # 'Epoch'
                         self.data[key] = (epoch_name, in_data)
                     elif len(in_data) == 1:
-                        # Only provided a single number in iterable, make that
-                        # the input for all times
-                        self.data[key] = (epoch_name,
-                                          [in_data[0]] * len(self.index))
+                        # Only provided a single number in iterable.
+                        if key in self.variables:
+                            # If it already exists, assign as defined.
+                            self.data[key] = in_data
+                        else:
+                            # Otherwise broadcast over time.
+                            warnings.warn(' '.join(('Input for {:}'.format(key),
+                                                    'is a single value.',
+                                                    'Broadcast over epoch.')))
+                            self.data[key] = (epoch_name,
+                                              [in_data] * len(self.index))
```

Comment:
These new lines are uncovered in the unit tests.  Recommend testing that this warning is raised.

Reply:
Updated the unit tests via parametrize to catch this case.  We were missing the case where input data could be an iterable with a single value.  You can also use the test snippet above with the following
```
import numpy as np

# Update the value
inst['preset_val'] = np.array(3.0)
# Set a new variable
inst['new_val'] = np.array(3.0)
```
to manually check outside the unit tests.